### PR TITLE
[SPARK-29947][SQL][FOLLOWUP] Fix table lookup cache

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Fix a bug in https://github.com/apache/spark/pull/26589 , to make this feature work.

### Why are the changes needed?

This feature doesn't work actually.

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

new test

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @wangyum @HyukjinKwon @dongjoon-hyun

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TableLookupCacheSuite.scala [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.catalog.{CatalogDatabase, CatalogStorageFormat, CatalogTable, CatalogTableType, ExternalCatalog, InMemoryCatalog, SessionCatalog}
+import org.apache.spark.sql.catalyst.dsl.plans._
+import org.apache.spark.sql.connector.InMemoryTableCatalog
+import org.apache.spark.sql.connector.catalog.{CatalogManager, CatalogNotFoundException, Identifier, Table, V1Table}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types._
+
+class TableLookupCacheSuite extends AnalysisTest with Matchers {
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TableLookupCacheSuite.scala]: Thank you for adding this test suite, @cloud-fan .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. The test suite is very useful for validation. Thanks.
Merged to master.

All test passed. The R failure is a known incoming feasibility check flakiness. cc @viirya 
```
* checking CRAN incoming feasibility ...Error in .check_package_CRAN_incoming(pkgdir) : 
  dims [product 26] do not match the length of object [0]
```

**Comment by viirya** [comment]
hmm, is CRAN failure still there now? I got their reply that was fixed. This failure looks like 5hrs ago. If there is new failure, please let me know.

**Pull request closed without merging**